efi/efi.go

```diff
@@ -19,10 +19,50 @@
 
 package efi
 
+import "github.com/canonical/go-tpm2"
+
 const (
-	platformFirmwarePCR = 0 // SRTM, POST BIOS, and Embedded Drivers
-	driversAndAppsPCR   = 2 // UEFI Drivers and UEFI Applications
-	bootManagerCodePCR  = 4 // Boot Manager Code and Boot Attempts PCR
-	secureBootPCR       = 7 // Secure Boot Policy Measurements PCR
-	kernelConfigPCR     = 12
+	platformFirmwarePCR tpm2.Handle = 0 // SRTM, POST BIOS, and Embedded Drivers
+	driversAndAppsPCR   tpm2.Handle = 2 // UEFI Drivers and UEFI Applications
+	bootManagerCodePCR  tpm2.Handle = 4 // Boot Manager Code and Boot Attempts PCR
+	secureBootPolicyPCR tpm2.Handle = 7 // Secure Boot Policy Measurements PCR
+	kernelConfigPCR     tpm2.Handle = 12
```

Comment:
is this just convience or does the spec considers PCR number references also Handles?

Reply:
Yes, PCRs are just considered like any other resource in the TPM and are addressed by a handle, in this case, with `0x00` being the first octet for all PCRs, which is why they can often just be referenced by an integer, although at some point that has to get converted to a type that's based on uint32 (which is what `tpm2.Handle` is.

See, eg, how go-tpm2 takes a `ResourceContext` which is used to reference some TPM resource with a name and a handle (and contains a few other bits) for commands that operate on PCRs, such as https://pkg.go.dev/github.com/canonical/go-tpm2#TPMContext.PCRExtend, just like any other command that operates on a TPM resource.

And here's the first octet for the handle of all resource types: https://pkg.go.dev/github.com/canonical/go-tpm2@master#HandleType